src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleHoleESP.kt

```diff
@@ -91,10 +97,138 @@ object ModuleHoleESP : Module("HoleESP", Category.RENDER) {
+                                    transparentColor,
+                                    transparentColor,
+                                    baseColor
+                                )
+                            )
+                            drawGradientQuad(
+                                listOf(
+                                    Vec3(box.maxX, 0.0, box.minZ),
+                                    Vec3(box.maxX, glowHeight, box.minZ),
+                                    Vec3(box.maxX, glowHeight, box.maxZ),
+                                    Vec3(box.maxX, 0.0, box.maxZ),
+                                ),
+                                listOf(
+                                    baseColor,
+                                    transparentColor,
+                                    transparentColor,
+                                    baseColor
+                                )
+                            )
+                            drawGradientQuad(
+                                listOf(
+                                    Vec3(box.maxX, 0.0, box.maxZ),
+                                    Vec3(box.maxX, glowHeight, box.maxZ),
+                                    Vec3(box.minX, glowHeight, box.maxZ),
+                                    Vec3(box.minX, 0.0, box.maxZ),
+                                ),
+                                listOf(
+                                    baseColor,
+                                    transparentColor,
+                                    transparentColor,
+                                    baseColor
+                                )
+                            )
+                            drawGradientQuad(
+                                listOf(
+                                    Vec3(box.minX, 0.0, box.maxZ),
+                                    Vec3(box.minX, glowHeight, box.maxZ),
+                                    Vec3(box.minX, glowHeight, box.minZ),
+                                    Vec3(box.minX, 0.0, box.minZ),
+                                ),
+                                listOf(
+                                    baseColor,
+                                    transparentColor,
+                                    transparentColor,
+                                    baseColor
+
+                                )
+                            )
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+
     val movementHandler = handler<PlayerTickEvent> { event ->
         this.updateScanRegion()
     }
 
+    private fun calculateFade(pos: BlockPos) =
+        if(distanceFade == 0f)
+            1f
+        else
+
+            ((1 - max(
+                (player.pos.y - pos.y) / verticalDistance,
+                Vec3d(player.pos.x - pos.x, 0.0, player.pos.z - pos.z).length() / horizontalDistance
+            ))
+                / distanceFade)
+            .coerceIn(0.0, 1.0).toFloat()
```

Comment:
This is honestly a bit hard to read and understand what you wanted this code to do. What you could do in such a situation is extracting parts of the equation to a variable. For example:
```Kotlin
private fun calculateFade(pos: BlockPos): Float {
    if (distanceFade == 0f)
        return 1f

    val verticalDistanceFraction = (player.pos.y - pos.y) / verticalDistance
    val horizontalDistanceFraction = Vec3d(player.pos.x - pos.x, 0.0, player.pos.z - pos.z).length() / horizontalDistance

    val fade = (1 - max(verticalDistanceFraction, horizontalDistanceFraction)) / distanceFade
    
    return fade.coerceIn(0.0, 1.0).toFloat()
}
```
I think this style is way more comprehensive